tests/mongo/test_mongo.py

```diff
@@ -683,3 +684,26 @@ def test_get_cache_key_with_datasource(mongo_connector, mongo_datasource):
     assert mongo_connector.get_cache_key(datasource) == mongo_connector.get_cache_key(
         datasource_with_extra_parameters
     )
+
+
+def test_get_cache_key_with_datasource(mongo_connector, mongo_datasource):
+    """
+    Mongo queries can contain objects, like dates, that are not JSON serializable.
+    The cache key must be away of the type of these objects
+    """
+    datasource_with_date = mongo_datasource(
+        collection='test_col',
+        query={'domain': 'plop', 'date': datetime(2021, 10, 4)},
+        parameters={'DOMAIN': 'domain1', 'FOO': 'BAR'},
+    )
+    # should not fail
+    cache_key_date = mongo_connector.get_cache_key(datasource_with_date)
+
+    datasource_with_str = mongo_datasource(
+        collection='test_col',
+        query={'domain': 'plop', 'date': str(datetime(2021, 10, 4))},
+        parameters={'DOMAIN': 'domain1', 'FOO': 'BAR'},
+    )
+    cache_key_str = mongo_connector.get_cache_key(datasource_with_str)
+
+    assert cache_key_str != cache_key_date
```

Comment:
On veut vérifier quoi avec cette assertion ? Que la date est sérialisée avec autre chose que `str` ?

Reply:
A query with a stringified date and query with an actual date should not yield the same cache key.
This is because there two queries will lead to different results in mongo: `find({ date: ISODate(2021, 10, 4)})` and `find({ date: '2021-10-04 00:00:00})`.